core/src/main/java/com/scalar/db/api/LikeExpression.java

```diff
@@ -0,0 +1,152 @@
+package com.scalar.db.api;
+
+import com.google.common.base.MoreObjects;
+import com.scalar.db.io.TextColumn;
+import java.util.Objects;
+import java.util.regex.Pattern;
+import javax.annotation.Nonnull;
+import javax.annotation.concurrent.Immutable;
+
+@Immutable
+public class LikeExpression extends ConditionalExpression {
+  private static final String DEFAULT_ESCAPE_CHAR = "\\";
+  @Nonnull private final String escape;
+  @Nonnull private final String regexPattern;
+
+  /**
+   * Constructs a {@code LikeExpression} with the specified column and operator.
+   *
+   * @param column a target column used to compare
+   * @param operator an operator used to compare the target column
+   */
+  LikeExpression(TextColumn column, Operator operator) {
+    this(column, operator, DEFAULT_ESCAPE_CHAR);
+  }
+
+  /**
+   * Constructs a {@code LikeExpression} with the specified column, operator and escape character.
+   *
+   * @param column a target column used to compare
+   * @param operator an operator used to compare the target column
+   * @param escape an escape character for the like operator
+   */
+  LikeExpression(TextColumn column, Operator operator, String escape) {
```

Comment:
It seems `escape` is basically used as a single length string (or empty.) How about simply treating it as a nullable character? It would work as same as empty string if it's null.

```java
  @Nullable private final Character escape;
```

Reply:
That's a good point. I also considered using `Character` as you suggested. But, in that case, users need to pass `null` to disable the escape character, right? (e.g., `.isLikeText("%xyz", null)`) This is because we want to use the default escape char `\` for `.isLikeText("%xyz")`. I prefer passing an empty string "" rather than null since its semantics is the same as SQL. What do you think?

Reply:
> users need to pass null to disable the escape character, right? (e.g., .isLikeText("%xyz", null)) 
Yes.

I understand your point to use consistent semantics with SQL. I think I was confused a bit because `com.scalar.db.storage.jdbc.RdbEngineStrategy#getEscape` returns null while `com.scalar.db.api.LikeExpression#getEscape` returns an empty. How about making `RdbEngineStrategy#getEscape` return an empty as well as `LikeExpression#getEscape`?

Reply:
After reading your comment https://github.com/scalar-labs/scalardb/pull/1046#discussion_r1311219596, I think my suggestion above was wrong... Please ignore it.

Reply:
> I prefer passing an empty string "" rather than null since its semantics is the same as SQL.
👍 

Maybe https://github.com/scalar-labs/scalardb/pull/1046#discussion_r1311085517 is also related. It would be great if you take care of it.

Reply:
> Maybe https://github.com/scalar-labs/scalardb/pull/1046#discussion_r1311085517 is also related. It would be great if you take care of it.

Sure. I will revise it. Thanks for the feedback.